pkg/chartverifier/profiles/profile.go

```diff
@@ -0,0 +1,100 @@
+package profiles
+
+import (
+	"errors"
+	"github.com/redhat-certification/chart-verifier/pkg/chartverifier/checks"
+	"gopkg.in/yaml.v3"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"runtime"
+)
+
+type Annotation string
+
+const (
+	DigestAnnotation                 Annotation = "Digest"
+	OCPVersionAnnotation             Annotation = "OCPVersion"
+	LastCertifiedTimestampAnnotation Annotation = "LastCertifiedTimestamp"
+)
+
+type Profile struct {
+	Apiversion  string          `json:"apiversion" yaml:"apiversion"`
+	Kind        string          `json:"kind" yaml:"kind"`
+	Name        string          `json:"name" yaml:"name"`
+	Annotations []Annotation    `json:"annotations" yaml:"annotations"`
+	Checks      []*ProfileCheck `json:"checks" yaml:"checks"`
+}
+
+type ProfileCheck struct {
+	Name string           `json:"name" yaml:"name"`
+	Type checks.CheckType `json:"type" yaml:"type"`
+}
+
+type FilteredRegistry map[checks.CheckName]checks.Check
+
+var profile *Profile
+
+func GetProfile() *Profile {
+
+	if profile != nil {
```

Comment:
Isn't `profile` always `nil` in this case?

Reply:
Good spot, in the rework I forgot to add back the code to set the profile name in the report. verifierBuilder now also calls ```profiles.GetProfile().``` With two callers I don't want to read and process the yaml file twice.